### ISSUE TITLE
HCM: Apply color changes on hover in High Contrast Mode for Search Input, `post-closebutton`, `post-togglebutton` and Chip components

### ISSUE THREAD

**Issue by alionazherdetska** [swisspost/design-system#4286]
**Description:**
The hover state color changes are missing for the following components, affecting accessibility:

1. Search Form Input: The search icon does not change color when hovering over the input field.
2. Close Button, Chip, Button Toggle : The color of the components does not change on hover.

**Steps to Reproduce:**
- Navigate to the pages of the following components:
    Search Input
    Button Close
    Button Toggle
    Chip
- Hover over each component and observe the behavior.
- Observe the lack of color changes during the hover state.
